R/coh_plot.R

```diff
@@ -0,0 +1,324 @@
+#' @examples
+#' data("cohortdata")
+#' start_cohort <- as.Date("2044-01-01")
+#' end_cohort <- as.Date("2044-12-31")
+#' date_interval <- c(start_cohort, end_cohort)
+#' plot_coverage(
+#'   data = cohortdata,
+#'   vacc_date_col = "vaccine_date_1",
+#'   unit = "month",
+#'   doses_count_color = "steelblue",
+#'   coverage_color = "mediumpurple",
+#'   date_interval = date_interval,
+#'   cumulative = FALSE
+#' )
+#' @export
+plot_coverage <- function(data,
+                          vacc_date_col,
+                          unit = c("day", "month", "year"),
+                          doses_count_color = "steelblue",
+                          coverage_color = "mediumpurple",
+                          date_interval = FALSE,
+                          cumulative = FALSE) {
+  checkmate::assert_data_frame(
+    data,
+    min.rows = 1L
+  )
+  checkmate::assert_character(
+    vacc_date_col,
+    min.len = 1L
+  )
+  checkmate::assert_names(
+    colnames(data),
+    must.include = c(vacc_date_col)
+  )
+  checkmate::assert_date(
+    data[[vacc_date_col]]
+  )
+  unit <- match.arg(unit, several.ok = FALSE)
+  checkmate::assert_string(
+    unit
+  )
+  checkmate::assert_logical(
+    cumulative, len = 1
+  )
+  checkmate::assert_character(
+    c(doses_count_color, coverage_color)
+  )
+
+  coverage <- coh_coverage(data = data,
+    vacc_date_col = vacc_date_col,
+    unit = unit,
+    date_interval = date_interval
+  )
+  if (cumulative) {
+    coverage$dose_plot <- coverage$cum_dose
+  } else {
+    coverage$dose_plot <- coverage$dose
+  }
+
+  colors <- c("c1" = doses_count_color, "c2" = coverage_color)
+  lbls <- c("c1" = "Doses", "c2" = "Coverage")
+  plt <- ggplot2::ggplot() +
+    ggplot2::geom_bar(ggplot2::aes(x = coverage$date,
+        y = coverage$dose_plot,
+        fill = "c1"
+      ), stat = "identity", alpha = 0.6
+    ) +
+    ggplot2::geom_line(ggplot2::aes(x = coverage$date,
+        y = coverage$coverage * max(coverage$dose_plot),
+        color = "c2", group = 1
+      ), linewidth = 1.3, linetype = 2
+    ) +
+    ggplot2::xlab("") +
+    ggplot2::ylab(paste0("Doses per ", unit)) +
+    ggplot2::scale_y_continuous(
+      labels = scales::label_number(scale_cut = scales::cut_short_scale()),
+      sec.axis = ggplot2::sec_axis(~. / max(coverage$dose_plot),
+        labels = scales::percent,
+        name = "Percentage of coverage"
+      )
+    ) +
+    ggplot2::scale_fill_manual(name = "",
+      values = colors,
+      labels = lbls
+    ) +
+    ggplot2::scale_color_manual(name = "",
+      values = colors,
+      labels = lbls
+    ) +
+    ggplot2::theme_classic() +
+    ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 90, hjust = 0),
+                   axis.text = ggplot2::element_text(size = 13),
+                   axis.title = ggplot2::element_text(size = 15),
+                   legend.title = ggplot2::element_text(size = 15),
+                   legend.text = ggplot2::element_text(size = 13))
```

Comment:
I would also suggest leaving these additions to the user.

I would suggest moving the legend to the top though, as it otherwise squashes the figure horizontally.

Reply:
Removed, see https://github.com/epiverse-trace/vaccineff/pull/31/commits/cbaf081d7d7f0a2792685563bb7e17893d3fe91f